packages/lib/slots.ts

```diff
@@ -194,7 +194,7 @@ function buildSlotsWithDateRanges({
 
     slotStartTime = slotStartTime.add(offsetStart ?? 0, "minutes").tz(timeZone);
 
-    while (!slotStartTime.add(eventLength, "minutes").subtract(1, "second").utc().isAfter(rangeEnd)) {
```

Comment:
We don't have to convert it into the UTC as it doesn't changes the absolute time, just changes from where we are observing the time.